src/main/scala/viper/gobra/frontend/info/implementation/resolution/NameResolution.scala

```diff
@@ -301,9 +307,51 @@ trait NameResolution { this: TypeInfoImpl =>
           findField(litType, id).getOrElse(UnknownEntity())
         } else symbTableLookup(n) // otherwise it is just a variable
 
+      case n: PIdnUse if tryEnclosingClosureImplementationProof(n).nonEmpty => entityWithinClosureImplementationProof(n)
+
       case n => symbTableLookup(n)
     }
 
+  private def entityWithinClosureImplementationProof(id: PIdnUse): Entity = {
+    val proof = tryEnclosingClosureImplementationProof(id).get
+    val (func, fBody, context) = proof.impl.spec.func match {
+      case PNamedOperand(id) => symbTableLookup(id) match {
+        case f: Function => (f, f.decl.body, f.context)
+        case c: Closure => (c, c.lit.decl.decl.body, c.context)
+        case _ => return UnknownEntity()
+      }
+      case PDot(base: PNamedOperand, id) =>
+        val pkg = symbTableLookup(base.id)
+        pkg match {
+          case pkg: Import =>
+            tryPackageLookup(RegularImport(pkg.decl.importPath), id, pkg.decl) match {
+              case Some((f: Function, _)) => (f, f.decl.body, f.context)
+              case _ => return UnknownEntity()
+            }
+          case _ => return UnknownEntity()
+        }
+      case _ => return UnknownEntity()
+    }
+
+    def addressable: Boolean = fBody.exists(_._1.shareableParameters.exists(_.name == id.name))
```

Comment:
this code looks quite complicates. Could you not call invoke on `proof.impl.spec.func` and from there extract the symbol. Also, the in and out paramters of a proof are always exclusive, i.e. the addressable method should always return false.

Reply:
Calling `resolve` on `proof.impl.spec.func` does not work because it causes a circular dependency (the method is called within `entity`)

I have now set the addressability of all parameters to `false`

Reply:
I have now split the method into 2 simpler ones, and made some other small changes